packages/chain-helpers/src/blockchain/Blockchain.ts

```diff
@@ -167,26 +173,71 @@ export async function submitSignedTx(
 
 export const dispatchTx = submitSignedTx
 
+const metadataHashes = new Map<string, HexString>()
+
+// Returns the Merkle root of the metadata as stored in the `ConfigService` cache. If not present, it computes it, stores it in the cache for future retrievals, and returns it.
+async function getMetadataHash(api: ApiPromise): Promise<HexString> {
+  const metadata = await api.call.metadata.metadataAtVersion(15)
```

Comment:
No need to fetch this if we have a cache hit, move down so it's skipped in that case

Reply:
https://github.com/KILTprotocol/sdk-js/pull/924#discussion_r1912965775